src/views/payment-sheet-views/card-view.js

```diff
@@ -614,32 +609,26 @@ CardView.prototype.onSelection = function () {
   }
 };
 
-CardView.prototype._hideUnsupportedCardIcons = function () {
-  var supportedCardTypes = this.client.getConfiguration().gatewayConfiguration.creditCards.supportedCardTypes;
-
-  Object.keys(constants.configurationCardTypes).forEach(function (paymentMethodCardType) {
-    var cardIcon;
-    var configurationCardType = constants.configurationCardTypes[paymentMethodCardType];
-
-    if (supportedCardTypes.indexOf(configurationCardType) === -1) {
-      cardIcon = this.getElementById(paymentMethodCardType + '-card-icon');
-      classList.add(cardIcon, 'braintree-hidden');
-    }
-  }.bind(this));
-};
+CardView.prototype._showSupportedCardIcons = function () {
+  var self = this;
 
-CardView.prototype._isCardTypeSupported = function (cardType) {
-  var configurationCardType = constants.configurationCardTypes[cardType];
-  var supportedCardTypes = this.client.getConfiguration().gatewayConfiguration.creditCards.supportedCardTypes;
+  this.hostedFieldsInstance.getSupportedCardTypes().then(function (supportedCardTypes) {
+    Object.keys(constants.configurationCardTypes).forEach(function (paymentMethodCardType) {
+      var cardIcon;
+      var configurationKey = constants.configurationCardTypes[paymentMethodCardType];
 
-  return supportedCardTypes.indexOf(configurationCardType) !== -1;
+      if (supportedCardTypes.indexOf(paymentMethodCardType) > -1) {
+        cardIcon = self.getElementById(configurationKey + '-card-icon');
+        classList.remove(cardIcon, 'braintree-hidden');
+      }
+    });
+  });
 };
 
 CardView.isEnabled = function (options) {
-  var gatewayConfiguration = options.client.getConfiguration().gatewayConfiguration;
   var disabledByMerchant = options.merchantConfiguration.card === false;
 
-  return Promise.resolve(!disabledByMerchant && gatewayConfiguration.creditCards.supportedCardTypes.length > 0);
```

Comment:
Does the check for supported cards here happen elsewhere now? Or does the actual behavior of `CardView.isEnabled` change without checking for supported cards

Reply:
We'll add a todo about figuring out the best way to disable the card view if we find out that the merchant is not enabled for credit cards.